Error while checking build/O2/fullCI for 52e8d4f1ce64bc87b6366184c693822ccfa72826 at 2023-04-21 18:21:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/11206-slc8_x86-64/0/Detectors/AOD/src/AODProducerWorkflowSpec.cxx:1744:14: error: no match for call to '(o2::framework::TableBuilder::persist<int, std::vector<float, std::allocator<float> >, std::vector<unsigned char, std::allocator<unsigned char> >, std::vector<float, std::allocator<float> >, std::vector<float, std::allocator<float> >, std::vector<unsigned char, std::allocator<unsigned char> > >(const std::array<const char*, 6>&)::<lambda(unsigned int, o2::framework::BuilderMaker<int>::FillType, o2::framework::BuilderMaker<std::vector<float> >::FillType, o2::framework::BuilderMaker<std::vector<unsigned char> >::FillType, o2::framework::BuilderMaker<std::vector<float> >::FillType, o2::framework::BuilderMaker<std::vector<float> >::FillType, o2::framework::BuilderMaker<std::vector<unsigned char> >::FillType)>) (int, int&, std::array<float, 26>::value_type&, std::array<float, 26>::value_type&, std::array<float, 26>::value_type&, std::array<float, 26>::value_type&, std::array<float, 26>::value_type&, std::array<float, 26>::value_type&, float [4], float [4], float [4], float [4], std::array<float, 10>::value_type&, std::array<float, 10>::value_type&, std::array<float, 10>::value_type&, std::array<float, 10>::value_type&, std::array<float, 10>::value_type&, std::array<float, 10>::value_type&)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11206/52e8d4f1ce64bc87b6366184c693822ccfa72826/build_O2_fullCI/pretty.html).

--- a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
+++ b/Detectors/AOD/src/AODProducerWorkflowSpec.cxx
@@ -1511,7 +1511,6 @@ void AODProducerWorkflowDPL::init(InitContext& ic)
     mFDDAmplitude = 0xFFFFFFFF;
     mT0Amplitude = 0xFFFFFFFF;
   }
-#ifdef O2_ZDC_NEWDATAMODEL
   // Initialize ZDC helper maps
   for (int ic = 0; ic < o2::zdc::NChannels; ic++) {
     mZDCEnergyMap[ic] = -std::numeric_limits<float>::infinity();
@@ -1519,7 +1518,6 @@ void AODProducerWorkflowDPL::init(InitContext& ic)
   for (int ic = 0; ic < o2::zdc::NTDCChannels; ic++) {
     mZDCTDCMap[ic] = -std::numeric_limits<float>::infinity();
   }
-#endif
 
   mTimer.Reset();
 }
@@ -1595,11 +1593,7 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
   auto& ambigMFTTracksBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "AMBIGUOUSMFTTR"});
   auto& ambigFwdTracksBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "AMBIGUOUSFWDTR"});
   auto& v0sBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "V0_001"});
-#ifdef O2_ZDC_NEWDATAMODEL
-  auto& zdcBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "ZDC_001"});
-#else
-  auto& zdcBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "ZDC"});
-#endif
+  auto& zdcBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "ZDC", 1});
   auto& caloCellsBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "CALO"});
   auto& caloCellsTRGTableBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "CALOTRIGGER"});
   auto& cpvClustersBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "CPVCLUSTER"});
@@ -1686,81 +1680,6 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
                fv0RecPoint.getTrigger().getTriggersignals());
   }
 
-#ifndef O2_ZDC_NEWDATAMODEL
-  for (auto zdcRecData : zdcBCRecData) {
-    uint64_t bc = zdcRecData.ir.toLong();
-    auto item = bcsMap.find(bc);
-    int bcID = -1;
-    if (item != bcsMap.end()) {
-      bcID = item->second;
-    } else {
-      LOG(fatal) << "Error: could not find a corresponding BC ID for a ZDC rec. point; BC = " << bc;
-    }
-    int fe, ne, ft, nt, fi, ni;
-    zdcRecData.getRef(fe, ne, ft, nt, fi, ni);
-    // initialize zdc helper maps
-    for (int ic = 0; ic < o2::zdc::NChannels; ic++) {
-      mZDCEnergyMap[ic] = -std::numeric_limits<float>::infinity();
-    }
-    for (int ic = 0; ic < o2::zdc::NTDCChannels; ic++) {
-      mZDCTDCMap[ic] = -std::numeric_limits<float>::infinity();
-    }
-    for (int ie = 0; ie < ne; ie++) {
-      auto& zdcEnergyData = zdcEnergies[fe + ie];
-      int ch = zdcEnergyData.ch();
-      float energy = zdcEnergyData.energy();
-      if (ch >= 0 && ch < o2::zdc::NChannels) {
-        mZDCEnergyMap[ch] = energy;
-      }
-    }
-    for (int it = 0; it < nt; it++) {
-      auto& tdc = zdcTDCData[ft + it];
-      int ch = tdc.ch();
-      float tdcValue = tdc.value();
-      if (ch >= 0 && ch < o2::zdc::NTDCChannels) {
-        mZDCTDCMap[ch] = tdcValue;
-      }
-    }
-    float energySectorZNA[4];
-    float energySectorZNC[4];
-    float energySectorZPA[4];
-    float energySectorZPC[4];
-    energySectorZNA[0] = mZDCEnergyMap[o2::zdc::IdZNA1];
-    energySectorZNA[1] = mZDCEnergyMap[o2::zdc::IdZNA2];
-    energySectorZNA[2] = mZDCEnergyMap[o2::zdc::IdZNA3];
-    energySectorZNA[3] = mZDCEnergyMap[o2::zdc::IdZNA4];
-    energySectorZNC[0] = mZDCEnergyMap[o2::zdc::IdZNC1];
-    energySectorZNC[1] = mZDCEnergyMap[o2::zdc::IdZNC2];
-    energySectorZNC[2] = mZDCEnergyMap[o2::zdc::IdZNC3];
-    energySectorZNC[3] = mZDCEnergyMap[o2::zdc::IdZNC4];
-    energySectorZPA[0] = mZDCEnergyMap[o2::zdc::IdZPA1];
-    energySectorZPA[1] = mZDCEnergyMap[o2::zdc::IdZPA2];
-    energySectorZPA[2] = mZDCEnergyMap[o2::zdc::IdZPA3];
-    energySectorZPA[3] = mZDCEnergyMap[o2::zdc::IdZPA4];
-    energySectorZPC[0] = mZDCEnergyMap[o2::zdc::IdZPC1];
-    energySectorZPC[1] = mZDCEnergyMap[o2::zdc::IdZPC2];
-    energySectorZPC[2] = mZDCEnergyMap[o2::zdc::IdZPC3];
-    energySectorZPC[3] = mZDCEnergyMap[o2::zdc::IdZPC4];
-    zdcCursor(0,
-              bcID,
-              mZDCEnergyMap[o2::zdc::IdZEM1],
-              mZDCEnergyMap[o2::zdc::IdZEM2],
-              mZDCEnergyMap[o2::zdc::IdZNAC],
-              mZDCEnergyMap[o2::zdc::IdZNCC],
-              mZDCEnergyMap[o2::zdc::IdZPAC],
-              mZDCEnergyMap[o2::zdc::IdZPCC],
-              energySectorZNA,
-              energySectorZNC,
-              energySectorZPA,
-              energySectorZPC,
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZEM1],
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZEM2],
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZNAC],
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZNCC],
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZPAC],
-              mZDCTDCMap[o2::zdc::TDCChannelID::TDCZPCC]);
-  }
-#else
   std::vector<float> zdcEnergy, zdcAmplitudes, zdcTime;
   std::vector<uint8_t> zdcChannelsE, zdcChannelsT;
   for (auto zdcRecData : zdcBCRecData) {
@@ -1798,7 +1717,6 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
               zdcTime,
               zdcChannelsT);
   }
-#endif
 
   // keep track event/source id for each mc-collision
   // using map and not unordered_map to ensure
@@ -2678,11 +2596,7 @@ DataProcessorSpec getAODProducerWorkflowSpec(GID::mask_t src, bool enableSV, boo
   outputs.emplace_back(OutputLabel{"O2ambiguousMFTtrack"}, "AOD", "AMBIGUOUSMFTTR", 0, Lifetime::Timeframe);
   outputs.emplace_back(OutputLabel{"O2ambiguousFwdtrack"}, "AOD", "AMBIGUOUSFWDTR", 0, Lifetime::Timeframe);
   outputs.emplace_back(OutputLabel{"O2v0_001"}, "AOD", "V0_001", 0, Lifetime::Timeframe);
-#ifdef O2_ZDC_NEWDATAMODEL
-  outputs.emplace_back(OutputLabel{"O2zdc_001"}, "AOD", "ZDC", 1, Lifetime::Timeframe);
-#else
-  outputs.emplace_back(OutputLabel{"O2zdc"}, "AOD", "ZDC", 0, Lifetime::Timeframe);
-#endif
+  outputs.emplace_back(OutputLabel{"O2zdc"}, "AOD", "ZDC", 1, Lifetime::Timeframe);
   outputs.emplace_back(OutputLabel{"O2caloCell"}, "AOD", "CALO", 0, Lifetime::Timeframe);
   outputs.emplace_back(OutputLabel{"O2caloCellTRGR"}, "AOD", "CALOTRIGGER", 0, Lifetime::Timeframe);
   outputs.emplace_back(OutputLabel{"O2cpvCluster"}, "AOD", "CPVCLUSTER", 0, Lifetime::Timeframe);

--- a/Framework/Core/include/Framework/AnalysisDataModel.h
+++ b/Framework/Core/include/Framework/AnalysisDataModel.h
@@ -11,8 +11,6 @@
 #ifndef O2_FRAMEWORK_ANALYSISDATAMODEL_H_
 #define O2_FRAMEWORK_ANALYSISDATAMODEL_H_
 
-// #define O2_ZDC_NEWDATAMODEL -> commented for now
-
 #include "Framework/ASoA.h"
 #include <cmath>
 #include <bitset>
@@ -1036,11 +1034,7 @@ DECLARE_SOA_TABLE_VERSIONED(Zdcs_001, "AOD", "ZDC", 1, //! ZDC information, vers
                             zdc::DyAmplitudeZEM1<zdc::ChannelT, zdc::Amplitude>, zdc::DyAmplitudeZEM2<zdc::ChannelT, zdc::Amplitude>,
                             zdc::DyAmplitudeZNA<zdc::ChannelT, zdc::Amplitude>, zdc::DyAmplitudeZNC<zdc::ChannelT, zdc::Amplitude>,
                             zdc::DyAmplitudeZPA<zdc::ChannelT, zdc::Amplitude>, zdc::DyAmplitudeZPC<zdc::ChannelT, zdc::Amplitude>); //
-#ifdef O2_ZDC_NEWDATAMODEL
 using Zdcs = Zdcs_001; //! new version
-#else
-using Zdcs = Zdcs_000; //! old version
-#endif
 using Zdc = Zdcs::iterator;
 
 namespace fv0a

